src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -2568,14 +2568,18 @@ private boolean handleSpecialTTEFunctions(
 
     private String getSignatureForInstanceTTEFn(
         String propName, List<String> classTemplateTypeNames, FunctionType ftype) {
-      // If ftype is foo.bar.Promise.prototype.then, extract className as Promise.
-      String className = getUnqualifiedName(getNamespace(getNamespace(ftype.getDisplayName())));
+      // If ftype is foo.bar.Promise.prototype.then, extract className as ಠ_ಠ.clutz.foo.bar.Promise.
+      String className =
+          Constants.INTERNAL_NAMESPACE + "." + getNamespace(getNamespace(ftype.getDisplayName()));
       Iterator<String> templateTypeNames = classTemplateTypeNames.iterator();
       if (!templateTypeNames.hasNext()) return null;
 
       String templateVarName = templateTypeNames.next();
 
-      String classTemplatizedType = className + " < RESULT >";
+      // TODO(lucassloan): goog.Promise has bad types that are coerced to any, so explicitly emit any
+      // and fix when the callers have been fixed.
+      String classTemplatizedType =
+          className.equals("ಠ_ಠ.clutz.goog.Promise") ? " any" : className + " < RESULT >";
```

Comment:
unless I am missing something, there is no test for users of 'goog.Promise' only tests for the declaration?

Reply:
Correct.